### PR TITLE
Fill the gaps

### PR DESCRIPTION
**Pull request by pawelpanasewicz**

### CODE REVIEW

File: app/actions/GetJourneyActionRefiner.scala [hunk 1]
```diff
@@ -16,29 +16,36 @@
...
     implicit val r: Request[A] = request
     for {
-      journey <- journeyService.get()
-    } yield Right(new JourneyRequest(journey, request))
+      maybeJourney: Option[Journey] <- journeyConnector.findLatestJourneyBySessionId()
+    } yield maybeJourney match {
+      case Some(journey) => Right(new JourneyRequest(journey, request))
+      case None =>
+        logger.warn(s"No journey found for sessionId: [ ${hc.sessionId} ]")
+        Left(Results.ImATeapot) //todo what should this be, obviously not teapot error
```
Comment (pawelpanasewicz) [inline app/actions/GetJourneyActionRefiner.scala:45]: Maybe we should display content: "For you security we timed you out?"  Let's analyse it.

File: app/controllers/EPayeStartController.scala [hunk 2]
```diff
@@ -60,22 +58,26 @@ class EPayeStartController @Inject() (
...
-          idType         = "AOR",
-          regime         = TaxRegime.Epaye,
-          id             = ttp.DefaultTaxId,
-          showFinancials = true
-        )
-      } yield routeResponse(data)
-      case _ => throw new IllegalStateException("missing journey")
-    }
+  val ePayeStart: Action[AnyContent] = as.default { implicit request =>
+    Ok("Not implemented yet, wip")
```
Comment (pawelpanasewicz) [inline app/controllers/EPayeStartController.scala:62]: Yes, it's on me.

File: app/controllers/EpayeGovUkController.scala [hunk 3]
```diff
@@ -18,20 +18,42 @@ package controllers
...
+      journeyConnector.Epaye.startJourneyGovUk(SjRequest.Epaye.Empty())
+    } else {
+      journeyConnector.Epaye.startJourneyDetachedUrl(SjRequest.Epaye.Empty())
+    }
+
+    sjResponse.map(r => Redirect(r.nextUrl.value))
+  }
+
+  /**
+   * Based on the "Referrer" http header it determines
```
Comment (pawelpanasewicz) [inline app/controllers/EpayeGovUkController.scala:52]: Btw, a "referer" is misspelled. It's misspelled in the RCF LOL

File: app/controllers/LandingController.scala [hunk 4]
```diff
@@ -22,18 +22,21 @@ import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
...
+) extends FrontendController(mcc)
   with Logging {
 
-  val landingPage: Action[AnyContent] = as.default.async { implicit request =>
+  val landingPage: Action[AnyContent] = as.default.async { _ =>
     Future.successful(Redirect(routes.EPayeStartController.ePayeLanding()))
   }
 
+  val tempStartPage: Action[AnyContent] = as.getJourney.async { implicit request =>
+    Future.successful(Ok(s"Journey Started: [ ], \n landing page/routes is in progress: SUPP-304, https://jira.tools.tax.service.gov.uk/browse/SUPP-304"))
```
Comment (pawelpanasewicz) [inline app/controllers/LandingController.scala:39]: ok

File: app/models/MockJourney.scala [hunk 5]
```diff
@@ -30,3 +31,19 @@ final case class MockJourney(
 object MockJourney {
   implicit val format: OFormat[Journey] = Json.format[Journey]
 }
+//todo this whole file needs to be deleted, but this change is already massive... I've put this in just to make the rest compile where mockJourney is used...
```
Comment (pawelpanasewicz) [inline app/models/MockJourney.scala:34]: ok

**Review by gitwojciech — APPROVED** [review]

**Pull request merged**